lib/SauceLabs.js

```diff
@@ -268,39 +268,22 @@ function generateLink(hostname, auth, date, job) {
 }
 
 function makeRequest(options, body, callback) {
-  var request = https.request(options, function (response) {
-    var result = '';
-
-    response
-      .on('data', function (chunk) {
-        result += chunk;
-      })
-      .on('end', function () {
-        var res;
-
-        try {
-          res = JSON.parse(result);
-        } catch (e) {
-          callback('Could not parse response: ' + result);
-          return;
-        }
-
-        if (response.statusCode === 200) {
-          callback(null, res);
-        } else {
-          callback(res);
-        }
-      });
-  });
+  request(options, function (error, response, body) {
 
-  request
-    .on('error', function (err) {
-      callback('Could not send request: ' + err.message);
-    });
+    if(error){
+      callback('Could not send request: ' + error.message);
+    }
 
-  if (body != null) {
-    request.write(body);
-  }
+    if (response.statusCode === 200) {
 
-  request.end();
+      callback(null, body);
+    } else {
+      console.log(typeof(body));
+      if(typeof(body) === 'string'){
+        callback('Could not parse response: ' + body);
```

Comment:
for example, if saucelabs returned with a 201, there would be no body but it would still be successful ?